pandas/core/arrays/interval.py

```diff
@@ -543,33 +543,87 @@ def __eq__(self, other):
             if is_interval_dtype(other_dtype):
                 if self.closed != other.categories.closed:
-                    return np.zeros(len(self), dtype=bool)
+                    return invalid_comparison(self, other, op)
                 other = other.categories.take(other.codes)
 
         # interval-like -> need same closed and matching endpoints
         if is_interval_dtype(other_dtype):
             if self.closed != other.closed:
-                return np.zeros(len(self), dtype=bool)
-            return (self._left == other.left) & (self._right == other.right)
+                return invalid_comparison(self, other, op)
+            if isinstance(other, Interval):
+                other = type(self)._from_sequence([other])
+                if self._combined.dtype.kind in ["m", "M"]:
+                    # Need to repeat bc we do not broadcast length-1
+                    # TODO: would be helpful to have a tile method to do
+                    #  this without copies
+                    other = other.repeat(len(self))
+            else:
+                other = type(self)(other)
+
+            if op is operator.eq:
+                return (self._combined[:, 0] == other._left) & (
+                    self._combined[:, 1] == other._right
+                )
+            elif op is operator.ne:
+                return (self._combined[:, 0] != other._left) | (
+                    self._combined[:, 1] != other._right
+                )
+            elif op is operator.gt:
+                return (self._combined[:, 0] > other._combined[:, 0]) | (
+                    (self._combined[:, 0] == other._left)
+                    & (self._combined[:, 1] > other._right)
+                )
+            elif op is operator.ge:
+                return (self == other) | (self > other)
+            elif op is operator.lt:
+                return (self._combined[:, 0] < other._combined[:, 0]) | (
+                    (self._combined[:, 0] == other._left)
+                    & (self._combined[:, 1] < other._right)
+                )
+            else:
+                # operator.lt
+                return (self == other) | (self < other)
 
         # non-interval/non-object dtype -> no matches
         if not is_object_dtype(other_dtype):
-            return np.zeros(len(self), dtype=bool)
+            return invalid_comparison(self, other, op)
 
         # object dtype -> iteratively check for intervals
-        result = np.zeros(len(self), dtype=bool)
-        for i, obj in enumerate(other):
-            # need object to be an Interval with same closed and endpoints
-            if (
-                isinstance(obj, Interval)
-                and self.closed == obj.closed
-                and self._left[i] == obj.left
-                and self._right[i] == obj.right
-            ):
-                result[i] = True
-
+        try:
+            result = np.zeros(len(self), dtype=bool)
```

Comment:
you should do the np.zeros outside of the try/except

Reply:
sure